### PR TITLE
Fix some lint

### PR DESCRIPTION
**Pull request by camjc**
Found using LGTM https://lgtm.com/projects/g/lazd/coronadatascraper/alerts/?mode=list

*Please review per commit*

There are still a few in find-features
image
but that function is pretty unwieldy so feel like we might want to simplify before removing those.

### CODE REVIEW

File: site/crossCheckReport.js [hunk 1]
```diff
@@ -35,8 +35,8 @@ function crossCheckReportTemplate(report) {
             <tr>
     `;
     const sourceURLShort = source.url.match(/^(?:https?:\/\/)?(?:[^@/\n]+@)?(?:www\.)?([^:/?\n]+)/)[1];
-    const curators = getContributors(source.curators, 'Curated by');
-    const sources = getContributors(source.sources, 'Sourced from');
```
Comment (lazd) [inline site/crossCheckReport.js]: Not lint, legit error -- looks like I screwed it up when I refactored.

File: site/lib/geography.js [hunk 2]
```diff
@@ -1,27 +1,8 @@
...
-    name += sep + location.country;
-    sep = ', ';
-  }
-  return name;
-};
+/** Get the full name of a location
+ * @param {{ city: string?; county: string?; state: string?; country: string?; }} location
+ */
+export const getName = location =>
+  [location.city, location.county, location.state, location.country].filter(Boolean).join(', ');
```
Comment (lazd) [inline site/lib/geography.js]: nice

File: src/events/processor/find-features/index.js [hunk 3]
```diff
@@ -104,6 +104,16 @@ function matchFeature(props, featuresData) {
...
 
+/**
+ * @param {string?} locationItem
+ * @param {{ properties: { name: string?; name_en: string?; region: string?; }; }} feature
+ */
+const doesLocationMatchFeature = (locationItem, feature) =>
+  locationItem &&
+  [feature.properties.name, feature.properties.name_en, feature.properties.region].some(
+    toMatch => toMatch === locationItem
+  );
```
Comment (lazd) [inline src/events/processor/find-features/index.js]: Nice, but I personally would rather see this as a proper `function` with `return` just so my brain doesn't hurt.

File: src/shared/scrapers/USA/NYT.js [hunk 4]
```diff
@@ -44,10 +44,6 @@ const scraper = {
       throw new Error(`Timeseries starts later than SCRAPE_DATE ${scrapeDateString}`);
     }
 
-    if (scrapeDate < firstDateInTimeseries) {
-      throw new Error(`Timeseries starts later than SCRAPE_DATE ${scrapeDateString}`);
-    }
```
Comment (lazd) [inline src/shared/scrapers/USA/NYT.js]: Never hurts to check twice, right? :P

File: src/events/processor/find-features/index.js (hunk 3 above)
Comment (lazd) [inline src/events/processor/find-features/index.js]:
```suggestion
function doesLocationMatchFeature(locationItem, feature) {
  return locationItem &&
  [feature.properties.name, feature.properties.name_en, feature.properties.region].some(
    toMatch => toMatch === locationItem
  );
}
```

Comment (camjc) [reply src/events/processor/find-features/index.js]: Why?
Assuming a good reason, we can enforce via linting if we want to.

**Pull request merged**